tests/main/refresh-devmode-nonproduction/task.yaml

```diff
@@ -0,0 +1,30 @@
+summary: Check that the refresh command works.
+
+systems: [-ubuntu-core-16-64, -ubuntu-core-16-device-amd64, -ubuntu-core-16-device-arm]
+
+details: |
+    These tests exercise the refresh command using different store backends.
+    The concrete store to be used is controlled with the STORE_TYPE environment
+    vairiable, if it is empty, a fake local store is used, if it has a value of
+    staginig or production the corresponding remote store is used.
+    When executing against the remote stores the tests rely in the existence of
+    a given snap with an updatable version (version string like 2.0+fake1) in the
+    edge channel.
+
+environment:
+    SNAP_NAME: test-snapd-tools
+    SNAP_VERSION_PATTERN: \d+\.\d+\+fake1
+    BLOB_DIR: $(pwd)/fake-store-blobdir
+    STORE_TYPE: fake
```

Comment:
Nope, the main intention of extracting the logic from the tasks is duplicating them, so that we can exclude the non-production tasks from being executed on the boards, they were giving assertion errors. According to @pedronis only the tests that are targeted to production should be executed in the devices.

We had previously variants in place for the different types of stores, if we recover them and move the logic back to the tasks we should somehow skip the non-production tests when executing in the boards, maybe I'm missing something, please let me know WDYT.